### PR TITLE
Update V8 and Hermes package versions

### PR DESCRIPTION
**Pull request by tudorms**
This includes updating the V8 engine to version 8.6 and the Hermes engine to version 0.7.1

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6400)

### CODE REVIEW

**Comment by NickGerleman** [comment]
@tudorms do you know if this fixes https://github.com/microsoft/react-native-windows/issues/6367? If so we can mark this PR as closing the issue, and switch one of the two Hermes jobs to x86.

**Review by Khalef1 — APPROVED** [review]

**Comment by NickGerleman** [comment]
We'll also want to update the bit in `packages\@react-native-windows\cli\src\generator-windows\index.ts`

```ts
  if (options.useHermes) {
    cppNugetPackages.push({
      id: 'ReactNative.Hermes.Windows',
      version: '0.6.3',
      hasProps: false,
      hasTargets: true,
    });
  }
```

**Review by NickGerleman — APPROVED** [review]

**Comment by tudorms** [comment]
Thanks for the tip, fixed now.

---
In reply to: [719835271](https://github.com/microsoft/react-native-windows/pull/6400#issuecomment-719835271) [](ancestors = 719835271)

**Comment by tudorms** [comment]
I don't know how to run the npx thing, but I did verify playground x86 release with Hermes 0.7.1 and basic boot scenario works.

---
In reply to: [719835027](https://github.com/microsoft/react-native-windows/pull/6400#issuecomment-719835027) [](ancestors = 719835027)

**Comment by NickGerleman** [comment]
Awesome. Linked this PR to close the issue tracking x86 issues. Consider updating one of the below in `.ado\windows-vs-pr.yml`  so that we get test coverage for x86. I can also do this as a follow-up if you want.

```yml
    strategy:
      # We should test x86 for one of these flavors once #6367 is resolved
      matrix:
        X64DebugCpp:
          language: cpp
          configuration: Debug
          platform: x64
        X64ReleaseCpp:
          language: cpp
          configuration: Release
          platform: x64
```

File: .ado/windows-vs-pr.yml [hunk 1]
```diff
@@ -662,7 +661,7 @@ jobs:
         X64ReleaseCpp:
```
Comment (NickGerleman) [inline .ado/windows-vs-pr.yml:661]: nit: need to rename this

**Pull request merged**